app/helpers/application_helper/toolbar/basic.rb

```diff
@@ -44,9 +42,43 @@ def separator
     {:separator => true}
   end
 
+  def definition
+    unless @loaded
+      load_overrides
+      @loaded = true
+    end
+
+    @definition
+  end
+
   private
 
+  def extension_classes
+    return [] if self.class.name.nil?
+
+    #Vmdb::Plugins.instance.vmdb_plugins.collect do |plugin|
+    Vmdb::Plugins.collect do |plugin|
+      name = plugin.root.join(
+        'app/toolbar_overrides',
+        self.class.name.to_s.split('::').last.underscore
+      )
+      name = "#{name}.rb"
+      if File.exist?(name)
+        require name
+        instance = [plugin.name.sub('::Engine',''), 'ToolbarOverrides', self.class.name.to_s.split('::').last].join('::')
+        instance.constantize
+      end
+    end.compact
+  end
+
+  def load_overrides
+    extension_classes.each do |klass|
+      @definition.merge!(klass.definition)
+    end
+  end
+
   def initialize
+    @loaded = false
     @definition = {}
```

Comment:
Unset both of these and let the accessor deal with the lazy loading on first access.